src/convert.rs

```diff
@@ -0,0 +1,285 @@
+
+    // Theoretically, it could be possible that input wires are also output wires in Garble.
+    // This, however, is strongly discouraged and not supported in Bristol fashion format since
+    // it reveals information about the input.
+    let input_wire_set: HashSet<_> = (0..total_input_gates).collect();
+    if output_gates.iter().any(|w| input_wire_set.contains(w)) {
+        return Err(ConverterError::OutputWireIsInput);
+    }
+
+    // Deal with duplicate output wires that is possible in Garble by simulating "identity" gates using two Xor
+    // gates, i.e., x XOR x XOR x = x, since the Bristol format requires every output bit to be a separate wire.
+    let mut seen = HashSet::new();
+    let mut wire_max = total_wires;
+    for out in &mut output_gates {
+        if !seen.insert(*out) {
+            mod_circuit.gates.push(Gate::Xor(*out, *out));
+            mod_circuit.gates.push(Gate::Xor(*out, wire_max));
+            *out = wire_max + 1;
+            wire_max += 2;
+            total_wires += 2;
+            total_gates += 2;
+        }
+    }
+
+    // Create the file that will contain the circuit described in Bristol fashion.
+    let mut file = File::create(path).map_err(ConverterError::FileWrite)?;
+    // The first line contains the number of parties, the number of wires, and the number of gates.
+    writeln!(file, "{} {}", total_gates, total_wires).map_err(ConverterError::FileWrite)?;
+    // The second line contains the number of parties, and the number of input wires per party.
+    write!(file, "{} ", mod_circuit.input_gates.len()).map_err(ConverterError::FileWrite)?;
+    for &input_len in &mod_circuit.input_gates {
+        write!(file, "{} ", input_len).map_err(ConverterError::FileWrite)?;
+    }
+    writeln!(file).map_err(ConverterError::FileWrite)?;
+    // The third line contains the number of output wires.
+    writeln!(file, "1 {}", output_gates.len()).map_err(ConverterError::FileWrite)?;
+    writeln!(file).map_err(ConverterError::FileWrite)?;
+
+    // In order to convert the circuit to the Bristol format, we need to create a mapping
+    // from the wires in the circuit to the wires in the Bristol format, where we need to make
+    // sure that the last wires contain the output in the right order.
+    let mut wires_map = vec![0; total_wires];
+    for (i, wire) in wires_map.iter_mut().take(total_input_gates).enumerate() {
+        *wire = i;
+    }
+
+    let mut out_idx = 0;
+    for (i, wire) in wires_map
+        .iter_mut()
+        .enumerate()
+        .take(total_wires)
+        .skip(total_input_gates)
+    {
+        if let Some(idx) = output_gates.iter().position(|&x| x == i) {
+            *wire = total_wires - output_gates.len() + idx;
+            out_idx += 1;
+        } else {
+            *wire = i - out_idx;
+        }
+    }
+
+    // Write the gates to the file in the Bristol format.
+    for (i, gate) in mod_circuit.gates.iter().enumerate() {
```

Comment:
Is `mod_circuit.gates` guaranteed to be in topological order? This is required for bristol circuits.